reframe/frontend/executors/__init__.py

```diff
@@ -168,15 +175,16 @@ def runall(self, checks, system):
             self._printer.timestamp('Started on', 'short double line')
             self._printer.info()
             self._runall(checks, system)
+            if self._max_retries:
+                self._retry_failed(checks, system)
+
         finally:
-            # Always update statistics and print the summary line
-            self._stats = self._policy.getstats()
-            num_failures = self._stats.num_failures()
```

Comment:
Keep this line so as to avoid the double call to `num_failures`. It will also facilitate the formatting.

Reply:
I really don't think that we should try to reduce the number of calls to functions of statistics, because (1) I do not believe that it is at all relevant for performance and (2) if it turned out to be performance relevant, then we should optimise the class statistics such that it stores once computed results internally to avoid recomputing them.

Reply:
My argument is mostly for clarity and easier formatting.

Reply:
Sorry, I thought you were still talking about the optimisation. Yes, of course it looks better with the additional line `num_failures = self._stats.num_failures()`. Done.